src/sbvr-api/errors.ts

```diff
@@ -10,5 +10,237 @@ export class InternalRequestError extends TypedError {}
+		super(404, error);
+	}
+}
+
+export class MethodNotAllowedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(405, error);
+	}
+}
+
+export class NotAcceptableError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(406, error);
+	}
+}
+
+export class ProxyAuthenticationRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(407, error);
+	}
+}
+
+export class RequestTimeoutError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(408, error);
+	}
+}
+
+export class ConflictError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(409, error);
+	}
+}
+
+export class GoneError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(410, error);
+	}
+}
+
+export class LengthRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(411, error);
+	}
+}
+
+export class PreconditionFailedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(412, error);
+	}
+}
+
+export class PayloadTooLargeError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(413, error);
+	}
+}
+
+export class URITooLongError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(414, error);
+	}
+}
+
+export class UnsupportedMediaTypeError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(415, error);
+	}
+}
+
+export class RequestedRangeNotSatisfiableError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(416, error);
+	}
+}
+
+export class ExpectationFailedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(417, error);
+	}
+}
+
+export class MisdirectedRequestError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(421, error);
+	}
+}
+
+export class UnprocessableEntityError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(422, error);
+	}
+}
+
+export class LockedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(423, error);
+	}
+}
+
+export class FailedDependencyError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(424, error);
+	}
+}
+
+export class UpgradeRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(426, error);
+	}
+}
+
+export class PreconditionRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(428, error);
+	}
+}
+
+export class TooManyRequestsError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(429, error);
+	}
+}
+
+export class RequestHeaderFieldsTooLargeError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(431, error);
+	}
+}
+
+export class UnavailableForLegalReasonsError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(451, error);
+	}
+}
+
+export class InternalServerError extends HttpError {
```

Comment:
There's an `InternalRequestError` which is the current equivalent, so it would be good to reuse it

Reply:
I was actually hesitant to reuse it, so that we don't have a breaking change where InternalRequestError suddenly starts passing it's message to the response.
After a quick search it seems that:
* In pine itself its only used once and w/o a message argument but it's exported.
* The API isn't using it at all (and I'm not aware of any other consumer).

Changing InternalRequestError to be an HttpError and dropping InternalServerError. Give me a note in case I misinterpreted the requested change here.

PS: TBH I liked the naming of InternalServerError more than InternalRequestError, it's more consistent with the http errors naming & other server libraries.